mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -163,47 +163,54 @@ class BinaryClassificationMetrics @Since("3.0.0") (
   private lazy val (
     cumulativeCounts: RDD[(Double, BinaryLabelCounter)],
     confusions: RDD[(Double, BinaryConfusionMatrix)]) = {
+
+    val numPartitions = if (numBins == 0) {
+      scoreLabelsWeight.getNumPartitions
+    } else {
+      numBins
```

Comment:
This value could be pretty large though. You might be repartitioning into 10000 partitions. I'm not sure this is worth it?

Reply:
In current impl, large input (with 10000 partitions) will resulit in curve with 10000 partition, and at least 10000 points.
In the pr, we regard one partition as one group, and there will be 1000 (by default) points.

Reply:
@srowen Another potential benefit is that, current impl can not deal with large group whose size is larger than `Int.MaxValue`, and if we use a partition to represent a gorup, this limit maybe overcomed since the block size limitation is eliminated.

Reply:
I agree, but now a normal-sized input with 100000 bins will result in 100000 partitions. That could really grind to a halt. Even in a more common case with a smallish number of points and smallish number of partitions and 1000 bins, 1000 partitions could simply be slower.

Reply:
@srowen   OK, I this this may go too far. I will close it.